marketplace-client/src/marketplace/src/app/system-accounts/search/search.component.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  Component,
+  OnInit,
+  Output,
+  EventEmitter,
+  ElementRef,
+} from '@angular/core';
+import { fromEvent } from 'rxjs';
+import { debounceTime, map, switchAll } from 'rxjs/operators';
+import { StaffMember } from 'src/app/_models/staff-member';
+import { UserDto } from 'src/app/_models/UserDto';
+import { SystemAccountService } from 'src/app/_services/system-account.service';
+
+@Component({
+  selector: 'app-search',
+  templateUrl: './search.component.html',
+  styleUrls: ['./search.component.css'],
+})
+export class SearchComponent implements OnInit {
+  //@Output() results: EventEmitter<User[]> = new EventEmitter<User[]>();
+  @Output() results: EventEmitter<any> = new EventEmitter<any>();
+  users: StaffMember[] = [];
+  search: string = '';
+  init: boolean = true;
+
+  constructor(private service: SystemAccountService, private el: ElementRef) {}
+
+  ngOnInit(): void {
+    this.search = this.service.getSearch();
+
+    fromEvent(this.el.nativeElement, 'keyup')
```

Comment:
it's safer and faster not to listen to `keyup` events, but to utilize angular's built-in event emitters. Any HTML input has a bunch of outputs from js-native events, like `(change)`, `(click)`, `(blur)`, etc. You can just use 1-way output binding to listen to them
(the current option is not an error, as well)